### PR TITLE
[ch158528] Avoid updating the same layer more than once when reordering widgets in Builder

### PR DESCRIPTION
**Pull request by Shylpx**
### Resources

- [Clubhouse story](https://app.clubhouse.io/cartoteam/story/158528/4global-admin-rate-limit-errors-reordering-builder-widgets)

### Context

- This case is very similar to https://github.com/CartoDB/cartodb/pull/16279. When widgets are being reordered, we are also triggering the `saveAnalysisForLayer` method for the layer linked with each widget. And when that method is called, a request is performed to update the layer's analysis nodes. So if there are layers with more than one widget, or there are complex analyses linking layers, the same request to update an analysis node will be performed multiple times. Then, Maps API will respond with a `429` (rate limit) error accordingly.
- In this case, the above is necessary but not enough. The main problem is that the number of widgets is not limited, so at some point, as many concurrent requests as widgets is too much and `nginx` blocks them.

### Changes

- Only trigger `saveAnalysisForLayer` once for each layer and save only its owned analysis.
- Send requests to save widgets in batches of 5 requests.

### CODE REVIEW

File: lib/assets/test/spec/builder/data/user-actions.spec.js [hunk 1]
```diff
@@ -2579,6 +2579,52 @@ describe('builder/data/user-actions', function () {
...
+        spyOn(this.b1, 'save');
+      });
+
+      it('should not save the layer analysis if the "saveLayerAnalysis" flag is set to false', function () {
+        this.userActions.saveWidget(this.wa1, false);
+
+        expect(this.analysisDefinitionsCollection.saveAnalysisForLayer).not.toHaveBeenCalled();
+      });
+
+      it('should save only the owned layer analysis if the "saveOnlyOwnedAnalysis" flag is set to true', function () {
```
Comment (VictorVelarde) [inline lib/assets/test/spec/builder/data/user-actions.spec.js:2622]: Looks like a nice approach

**Review by VictorVelarde — APPROVED** [review]
Nice work. Fix the conflicts and then ready to go 🚀

**Comment by Shylpx** [comment]
Hi @VictorVelarde! Could you please review the last commits? :pray: This fix was not enough, since the concurrent requests to save widgets (one per widget) are reaching the `nginx` limits, so they are being sent now in batches of 5 requests.

**Review by VictorVelarde — APPROVED** [review]
👍🏻  LGTM.

**Comment by VictorVelarde** [comment]
Update assets, and feel free to merge

**Pull request merged**